### PR TITLE
DOCSP-43085: Compound Indexes

### PR DESCRIPTION
**Pull request by mcmorisi**
# Pull Request Info

[PR Reviewing Guidelines](https://github.com/mongodb/docs-golang/blob/master/REVIEWING.md)

JIRA - <https://jira.mongodb.org/browse/DOCSP-43085>
Staging - <https://deploy-preview-46--docs-c.netlify.app/indexes/compound-index/>

## Self-Review Checklist

- [x] Is this free of any warnings or errors in the RST?
- [x] Did you run a spell-check?
- [x] Did you run a grammar-check?
- [x] Are all the links working?
- [x] Are the [facets and meta keywords](https://wiki.corp.mongodb.com/display/DE/Docs+Taxonomy) accurate?

### CODE REVIEW

File: source/indexes/compound-index.txt [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+
+Overview
+--------
+
+:manual:`Compound indexes </core/index-compound/>` hold references to multiple
+fields within a collection's documents, improving query and sort performance.
+
+When creating a compound index, you must specify the following:
+
+- The fields on which to create the index.
```
Comment (rustagir) [inline source/indexes/compound-index.txt:28]:
```suggestion
- The fields to index. 
```

File: source/indexes/compound-index.txt [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+   :values: reference
+ 
+.. meta::
+   :keywords: index, query, optimization, efficiency
+
+Overview
+--------
+
+:manual:`Compound indexes </core/index-compound/>` hold references to multiple
+fields within a collection's documents, improving query and sort performance.
```
Comment (rustagir) [inline source/indexes/compound-index.txt:24]: S: add a sentence like "Use the ``mongoc_collection_create_indexes_with_opts()`` method to create an index.

File: source/indexes/compound-index.txt [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+
+.. literalinclude:: /includes/indexes/indexes.c
+   :start-after: start-index-compound
+   :end-before: end-index-compound
+   :language: c
+   :copyable:
+   :dedent:
+
+The following is an example of a query that uses the index created in
+the preceding code sample:
```
Comment (rustagir) [inline source/indexes/compound-index.txt:54]:
```suggestion
The following example performs a query that uses the index created in
the preceding code:
```

**Review by rustagir — APPROVED** [review]
couple small things, can take another look if needed

**Review by kevinAlbs — APPROVED** [review]

**Pull request merged**